u2f-gae-demo/src/com/google/u2f/gaedemo/storage/TokenStorageData.java

```diff
@@ -53,18 +55,29 @@ public void updateCounter(int newCounterValue) {
 
   public SecurityKeyData getSecurityKeyData() {
     X509Certificate x509cert = parseCertificate(attestationCert);
-    return new SecurityKeyData(enrollmentTime, transports, keyHandle,
-        publicKey, x509cert, counter);
+    return new SecurityKeyData(enrollmentTime, transports, keyHandle, publicKey, x509cert, counter);
   }
 
   public JsonObject toJson() {
+    X509Certificate x509cert = getSecurityKeyData().getAttestationCertificate();
     JsonObject json = new JsonObject();
     json.addProperty("enrollment_time", enrollmentTime);
     json.add("transports", getJsonTransports());
     json.addProperty("key_handle", Hex.encodeHexString(keyHandle));
     json.addProperty("public_key", Hex.encodeHexString(publicKey));
-    json.addProperty("issuer",
-        getSecurityKeyData().getAttestationCertificate().getIssuerX500Principal().getName());
+    json.addProperty("issuer", x509cert.getIssuerX500Principal().getName());
+
+    try {
+      AndroidKeyStoreAttestation androidKeyStoreAttestation =
+          AndroidKeyStoreAttestation.Parse(x509cert);
+      if (androidKeyStoreAttestation != null) {
+        json.add("android_attestation", androidKeyStoreAttestation.toJson());
+      }
+
+    } catch (CertificateParsingException e) {
+      // no android attestation
```

Comment:
Wouldn't the "no android attestation" case result in null getting returned above? Wouldn't you at least want to log this exception, in case it's throwing for some other reason?